inc/classes/preload/class-sitemap.php

```diff
@@ -0,0 +1,126 @@
+<?php
+namespace WP_Rocket\Preload;
+
+/**
+ * Sitemap preload
+ *
+ * @since 3.2
+ * @author Remy Perona
+ */
+class Sitemap extends Abstract_Preload {
+	/**
+	 * Launches the sitemap preload
+	 *
+	 * @since 3.2
+	 * @author Remy Perona
+	 *
+	 * @param array $sitemaps Sitemaps to use for preloading.
+	 * @return void
+	 */
+	public function run_preload( $sitemaps ) {
+		$urls_group = [];
+
+		foreach ( $sitemaps as $sitemap_type => $sitemap_url ) {
+			/**
+			 * Fires before WP Rocket sitemap preload is called for a sitemap URL
+			 *
+			 * @since 2.8
+			 *
+			 * @param string $sitemap_type  the sitemap identifier
+			 * @param string $sitemap_url sitemap URL to be crawler
+			*/
+			do_action( 'before_run_rocket_sitemap_preload', $sitemap_type, $sitemap_url );
+
+			$urls_group[] = $this->process_sitemap( $sitemap_url );
+
+			/**
+			 * Fires after WP Rocket sitemap preload was called for a sitemap URL
+			 *
+			 * @since 2.8
+			 *
+			 * @param string $sitemap_type  the sitemap identifier
+			 * @param string $sitemap_url sitemap URL crawled
+			*/
+			do_action( 'after_run_rocket_sitemap_preload', $sitemap_type, $sitemap_url );
+		}
+
+		foreach ( $urls_group as $urls ) {
```

Comment:
```php
$urls_group = array_filter( $urls_group );

if ( ! $urls_group ) {
    return;
}
```

And then this can be removed:
```php
if ( empty( $urls ) ) {
    continue;
}
```